src/main/java/io/cdap/plugin/servicenow/source/ServiceNowSourceConfig.java

```diff
@@ -395,72 +212,44 @@ private void validateTableQueryMode(FailureCollector collector) {
     if (Util.isNullOrEmpty(tableName)) {
       collector.addFailure("Table name must be specified.", null)
         .withConfigProperty(ServiceNowConstants.PROPERTY_TABLE_NAME);
+    } else {
+      // Call API to fetch first record from the table
+      ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
```

Comment:
looks like we have the same config validation logic in ServiceNowMultiSourceCOnfig. Can we dedupe this?

Reply:
Yes, Removed the duplicates. Defined the validation logic in the base class: `ServiceNowBaseSourceConfig` [c5d6fbb](https://github.com/data-integrations/servicenow-plugins/pull/8/commits/c5d6fbb1b2d49b43e78aadd471064b92574c868e)